test/test_uwsgi.py

```diff
@@ -0,0 +1,65 @@
+import flask
+import mock
+import time
+
+from cdispyutils import uwsgi
+
+
+def test_user_harakiri():
+    app = flask.Flask("test")
+
+    @app.route("/")
```

Comment:
you can use the `app` from `test/conftest.py` and also the `client` fixture from `pytest-flask` (instead of `app.test_client()`)

Reply:
Hmm it took me some time to find out that, the `client` fixture calls `teardown_request` at teardown time of the test case (and twice!!), while we still need to assert that the timeout is actually cleared at `teardown_request`. Therefore, `client` fixture is not used.

There is no `test/conftest.py` in `cdispyutils` package, but I'm anyway making in-place fixtures and splitting tests. (Hit a side-effect of patching `sys.modules`, took some time debugging to find out the root cause)

Reply:
Ahh, sorry, I was looking at my local copy with a different branch I was working on that has a `conftest.py` with the app fixture. I'll rescind my earlier comment...sorry to give you bogus feedback :grimacing:

Reply:
No worries! It's always good to have better code. 😄

Reply:
This guy is a good read: https://github.com/pytest-dev/pytest-flask/blob/master/pytest_flask/fixtures.py

I'd take a look at liveserver, I haven't had enough time to dig deep in here but feels like it might be pertinent. 

client in pytest_flask requires an app fixture.

Reply:
@jbarno thx for the info! I assume you were addressing the first issue that `teardown_request` is called (twice) at test case tearDown time? The two calls are:

1. Yield fixture `client` [exiting](https://github.com/pytest-dev/pytest-flask/blob/master/pytest_flask/fixtures.py#L26) the PEP-343 context of Flask `test_client`;
2. Auto-used fixture `_push_request_context` [poping](https://github.com/pytest-dev/pytest-flask/blob/master/pytest_flask/plugin.py#L124) the request context at the tearDown time of fixture `request`.